pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -213,6 +239,79 @@ func (r *ReconcileUserAccount) ensureIdentity(logger logr.Logger, userAcc *toolc
 	return identity, false, nil
 }
 
+// setFinalizers sets the finalizers for UserAccount
+func (r *ReconcileUserAccount) setFinalizers(userAcc *toolchainv1alpha1.UserAccount) error {
+	// Add the finalizer if it is not present
+	if !containsString(userAcc.ObjectMeta.Finalizers, userAccFinalizerName) {
+		userAcc.ObjectMeta.Finalizers = append(userAcc.ObjectMeta.Finalizers, userAccFinalizerName)
+		if err := r.client.Update(context.TODO(), userAcc); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// deleteUser deletes the user resource
+func (r *ReconcileUserAccount) deleteUser(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (error, bool) {
+	// Get the User associated with the UserAccount
+	user := &userv1.User{}
+	userErr := r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+	if userErr != nil {
```

Comment:
If you don't mind, can we use `err` for any error unless we want to distinguish two different error in same scope.   I am asking as generally in our team `err` is used rather `<prefix>err<suffix>` and that's also idiomatic go way.  This is to make sure we have consistence coding style throughout our codebase.

Reply:
fixed here: 008771ba402bc3dafb894c1eb8d0a7851f07df0d